### PR TITLE
[WIP] Fix ImportError for scikit-learn >= 0.23

### PR DESCRIPTION
**Pull request by jwmueller**
Fixes issue:  https://github.com/scikit-optimize/scikit-optimize/issues/902

Without this fix, all packages that depend on scikit-optimize are now broken unless they manually downgrade their sklearn version.
Note that the sklearn MaskedArray was anyway only needed for outdated versions of numpy: 
https://github.com/scikit-learn/scikit-learn/blob/0.22.X/sklearn/utils/fixes.py#L159

### CODE REVIEW

**Comment by jwmueller** [comment]
All CI tests pass except for one using scikit-learn 0.21.3, but it's unclear why that one failed.  That version of scikit-learn contains the same MaskedArray code as all the other versions in the passed CI tests:   https://github.com/scikit-learn/scikit-learn/blob/0.21.3/sklearn/utils/fixes.py#L166

**Comment by betatim** [comment]
The tests that fail are related to comparing two arrays that should be the same/very similar but are wildly different now. My guess is that the calculations that lead to them make use of masked arrays somewhere :-/

**Comment by jwmueller** [comment]
@betatim I think there's an issue in the CI tests.  The same test fails when I just committed a "dummy test" version of the code with no change from master... 

This is the failing test:

DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY_VERSION="1.16.0" SCIPY_VERSION="1.0.0" PYAML_VERSION="17.8.0" SCIKIT_LEARN_VERSION="0.21.3" MATPLOTLIB_VERSION="*" COVERAGE="true" JOBLIB_VERSION="0.13"

**Comment by rth** [comment]
So the breakage is the result of https://github.com/scikit-learn/scikit-learn/pull/16725 which removed backports for numpy <1.12.0. The test that failed in the second commit did so for numpy 1.16.0, in which case,
```
from sklearn.utils.fixes import MaskedArray
```
should be exactly equivalent to,
```
from numpy.ma import MaskedArray
```
so the initial fix should indeed work.

**Comment by betatim** [comment]
@jwmueller can you push the commit with your proposed change (and maybe squash the "not used anymore" ones)?

CI with no changes fails because of the problem you are trying to fix here (I think) :)

**Comment by holgern** [comment]
Should we use the same dependecies as scikit-learn:

scikit-learn requires:

Python (>= 3.6)
NumPy (>= 1.13.3)
SciPy (>= 0.19.1)
joblib (>= 0.11)

**Comment by betatim** [comment]
I think increasing the minimum required version of scikit-learn to 0.21 (last two releases before the current one) would be good and reduce the amount of cross dependency compatibility issues. This will also raise the minimum required version of numpy, scipy and joblib. Or put differently: I wouldn't raise those other dependencies to above what scikit-learn requires.

**Comment by holgern** [comment]
@jwmueller Can you push the changes for MaskedArray again?

I will take care of the failing CI.

**Comment by holgern** [comment]
The CI is fixed in pr #906 
image

Waiting for your fix on MaskedArray :)

**Comment by holgern** [comment]
You can merge th upstream changes with:
https://github.com/jwmueller/scikit-optimize/compare/patch-1...scikit-optimize:master?expand=1
This should fix the CI.

**Comment by jwmueller** [comment]
@holgern @betatim I've re-added the changes and now the CI test for sklearn==0.21.3 passes where it failed before.  But your newly added CI test for sklearn==0.23.0 now fails.

Please feel free to take over and modify the code yourself.  It's probably not worth you waiting for my responses just to address these 3-line code changes...

**Comment by holgern** [comment]
The unit tests are broken due to changes in sklearn 0.23 on Gaussian processes regression: https://github.com/scikit-learn/scikit-learn/pull/15782

**Comment by holgern** [comment]
Has been fixed in #939
